src/foam/nanos/ruler/AsyncAgency.java

```diff
@@ -0,0 +1,19 @@
+package foam.nanos.ruler;
+
+import foam.core.Agency;
+import foam.core.ContextAgent;
+import foam.core.X;
+
+/**
+ * AsyncAgency is passed to async rule.action
+ * for immediate execution of agent.
+ */
+public class AsyncAgency implements Agency {
```

Comment:
I would name this class after what it does, rather than where you're currently using it. So, something like: DirectAgency, because it runs it directly.

See: https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/Executor.html